src/inmanta/ast/statements/define.py

```diff
@@ -290,12 +287,14 @@ def evaluate(self) -> None:
         """
         try:
             cls = self.namespace.get_type(self.entity)
+            self.anchors = [TypeReferenceAnchor(self.namespace, self.entity)]
             if not isinstance(cls, Entity):
                 raise TypingException(
                     self, "Implementation can only be define for an Entity, but %s is a %s" % (self.entity, cls)
                 )
             self.type.set_type(cls)
             self.copy_location(self.type)
+            self.anchors.extend(self.block.get_anchors())
```

Comment:
Oh damn, this does not work, does it? Afaik we only normalize the types (e.g. `Implementation`, which normalizes the block) after we've evaluated all definition statements. Perhaps we have to add a `normalize` stage to the `DefinitionStatement`?

Let me know if I'm missing something.